src/azure-cli/azure/cli/command_modules/vm/_vm_utils.py

```diff
@@ -173,7 +177,8 @@ def normalize_disk_info(image_data_disks=None,
             'lun': i,
             'managedDisk': {'storageAccountType': None},
             'createOption': 'empty',
-            'diskSizeGB': sizes_copy.pop(0)
+            'diskSizeGB': sizes_copy.pop(0),
+            'deleteOption': data_disk_delete_option if isinstance(data_disk_delete_option, str) else None
```

Comment:
If the type of `data_disk_delete_option` is dict, is it in line with the expectation that `deleteOption` will be None?

Reply:
It is a good question. For correct usage, it cannot be dict as in this case, a new disk will be created and customer cannot specify the disk delete option with key=value pair  because they have no idea about the key.

Reply:
OK, got it~